### ISSUE TITLE
Report connectionQuality value from useParticipant never improves

### ISSUE THREAD

**Issue by jbrowning** [livekit/client-sdk-react-native#64]
**Describe the bug**

`connectionQuality` returned by `useParticipant` will never recover from "poor" even if the connection improves.

Example session: https://cloud.livekit.io/projects/p_3tux3vr78ux/sessions/RM_mUW6ymhUZXo9/participants/PA_AZcvSwaWa2rb

**To Reproduce**

1. Observe the `connectionQuality` value returned by `useParticipant`
2. Wait for `connectionQuality` to degrade to poor
3. Wait for the affected client's actual connection quality to improve
3. See that the `connectionQuality` value will never change even if the actual connection quality changes

**Expected behavior**

Reported connection quality should improve as the client's connection improves. As you can in the screenshot below, the connection quality in the example session improves, framerate hovers around 30fps, but the improved connection quality is never reported to the publisher nor any subscribers.

**Screenshots**

![lkquality](https://user-images.githubusercontent.com/123726/235497834-d49b4188-97ca-465c-a611-9655a5f9854d.png)

**Device Info:**

- Device: Google Pixel 4, iPhone 6s, iPhone 14 Pro
- OS: Android 13 (Build TP1A.221005.002.B2), iOS 15.7.5, iOS 16.2

**Dependencies Info (please reference your package-lock.json or yarn.lock file):**

- @livekit/react-native: 1.0.0
- livekit-client: 1.8.0
- react-native-webrtc: 111.0.0

The issue also occurs in older versions such as:

- @livekit/react-native: github:livekit/client-sdk-react-native#5ccd65b01e4ea591e10dac6cd2334d2ff6b60e34
- livekit-client: 1.6.0
- react-native-webrtc: 106.0.0

**Additional context**

Manually subscribing to the `ParticipantEvent.ConnectionQualityChanged` event has the same issue.

**Comment by davidliu** [comment]
To be clear, `ParticipantEvent.ConnectionQualityChanged` doesn't fire an event when the connectionQuality improves?

Is it possible for you to see if the `participant.connectionQuality` variable remains poor (e.g. by periodically querying it)? Trying to locate where this issue is coming from. I can see that on the server side, the connection quality did recover.

**Comment by jbrowning** [comment]
Correct. In my case I listened to the `ParticipantEvent.ConnectionQualityChanged` event *and* observed the `connectionQuality` value from the `useParticipant` hook with `useEffect` in the same component and both reported the same "poor" value. Neither updated when the connection quality improved.

**Comment by davidliu** [comment]
Can you check whether the `connectionQuality` on the original Participant object changes or not (i.e. without the `useParticipant` hook)?

**Comment by jbrowning** [comment]
In our case `participant.connectionQuality` is always updated to the value received by the `ParticipantEvent.ConnectionQualityChanged` event callback. Changes seem to be consistently in this order:

1. `ParticipantEvent.ConnectionQualityChanged` event received
2. `connectionQuality` from `useParticipant` is updated
3. `participant.connectionQuality` is updated

**Comment by jbrowning** [comment]
@davidliu After additional testing it *appears* that Android clients are reporting improved connection quality as expected. This may be isolated to iOS publishers.

**Comment by jbrowning** [comment]
@davidliu It also looks like the "poor" connection quality threshold may be too sensitive.

Here's another example: https://cloud.livekit.io/projects/p_3tux3vr78ux/sessions/RM_WLgKDXUqcfrb/participants/PA_5oAde2vYxeMg

In this case the connection was reported as "poor" within a few seconds of connecting but the Quality/MOS metric never went below 3.0. Published FPS & bitrate were both stable during the entire session.

**Comment by davidzhao** [comment]
@jbrowning which version of the ReactNative SDK are you using? I think versions prior to 1.0 had a bug with simulcast publishing, which caused connection quality to drop to poor.

**Comment by davidliu** [comment]
@davidzhao looks like there's another issue on ios for simulcast that's causing the other layers not to be sent up, and causing the quality to drop when other subscribers request those layers.

**Comment by jbrowning** [comment]
@davidzhao We've seen this behavior on:

* livekit/client-sdk-react-native: github:livekit/client-sdk-react-native#5ccd65b01e4ea591e10dac6cd2334d2ff6b60e34
* @livekit/react-native: 1.0.0

I'll try disabling simulcast and see what happens.

**Comment by jbrowning** [comment]
I've confirmed that disabling simulcast on both `github:livekit/client-sdk-react-native#5ccd65b01e4ea591e10dac6cd2334d2ff6b60e34` and 1.0.0 seems to fix the issue.

**Comment by davidzhao** [comment]
@jbrowning Thanks for the confirmation. As @davidliu mentioned, the connection quality drop is due to a bug in 1.0 with iOS simulcast publishing. investigating.
